lib/modules/manager/composer/artifacts.ts

```diff
@@ -19,27 +19,44 @@ import {
 import { getRepoStatus } from '../../../util/git';
 import * as hostRules from '../../../util/host-rules';
 import { regEx } from '../../../util/regex';
+import { GitTagsDatasource } from '../../datasource/git-tags';
 import { PackagistDatasource } from '../../datasource/packagist';
 import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
 import type { AuthJson, ComposerLock } from './types';
 import {
   composerVersioningId,
   extractConstraints,
+  findGithubToken,
   getComposerArguments,
   getPhpConstraint,
+  isGithubPersonalAccessToken,
   requireComposerDependencyInstallation,
+  takePersonalAccessTokenIfPossible,
 } from './utils';
 
 function getAuthJson(): string | null {
   const authJson: AuthJson = {};
 
-  const githubCredentials = hostRules.find({
+  const githubToken = findGithubToken({
     hostType: PlatformId.Github,
     url: 'https://api.github.com/',
   });
-  if (githubCredentials?.token) {
+
+  const gitTagsGithubToken = findGithubToken({
+    hostType: GitTagsDatasource.id,
+    url: 'https://github.com',
+  });
+
+  const selectedGithubToken = takePersonalAccessTokenIfPossible(
+    githubToken,
+    gitTagsGithubToken
+  );
+  if (selectedGithubToken) {
+    if (!isGithubPersonalAccessToken(selectedGithubToken)) {
+      logger.debug(`Selected github token is not a Personal Access Token`);
+    }
```

Comment:
```suggestion
    if (isGithubPersonalAccessToken(selectedGithubToken)) {
      logger.debug('Using GitHub Personal Access Token');
    } else if (isGithubServerToServerToken(selectedGithubToken)) {
      logger.debug('Using GitHub Server-to-Server token');
    } else {
      logger.debug('Using unknown GitHub token type');
    }
```

Reply:
Good idea, I will add the fine grained token too